File: Week 2/id_545/LeetCode_242_545.js
// 242. 有效的字母异位词

// 排序 比较是否相等
// var isAnagram = function(s, t) {
//     return s.split('').sort().join('') == t.split('').sort().join('')
// };


// 哈希表
var isAnagram = function(s, t) {
    if (s.length != t.length) return false
    let dct = {}
    for(let i in s) {
        let m = s[i]
        dct[ m ] != undefined ? dct[ m ]++ : dct[ m ] = 1
    }

    for(let j in t) {
        let m = t[j]
        if (dct[ m ] != undefined) {
            dct[ m ]--
            if (dct[ m ] < 0) {
                return false
            }
        } else {
            return false
        }
    }
    return true
}

File: Week 2/id_545/LeetCode_242_545.py
## 242. 有效的字母异位词

# 排序 比较是否相等
# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         return sorted(s) == sorted(t)

# 哈希表
# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         if len(s) != len(t): return False
#         dct = {}
#         for i in s:
#             dct[i] = dct.get(i, 0) + 1
#         for j in t:
#             dct[j] = dct.get(j, 0) - 1
#             if dct[j] < 0: return False
#         return True

# 集合
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        t_set = set(t)
        if len(s) != len(t) or t_set != set(s): return False
        for i in t_set:
            if s.count(i) != t.count(i):
                return False
        return True